src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1500,6 +1512,11 @@ private DebugId GetLambdaId(SyntaxNode syntax, ClosureKind closureKind, int clos
 
         private void AddSynthesizedMethod(MethodSymbol method, BoundStatement body)
         {
+            if (body == null)
```

Comment:
>if (body == null) [](start = 12, length = 17)

Where do we make sure the method definition is added to the right type for metadata emit?

Reply:
In `RewriteLambdaOrLocalFunction` we call `PEModuleBuilder.AddSynthesizedDefinition` and then later call `PEModuleBuilder.GetSynthesizedMethods(TNamedTypeSymbol)` when emitting the metadata

---
In reply to: [354056587](https://github.com/dotnet/roslyn/pull/40011#discussion_r354056587) [](ancestors = 354056587)